### ISSUE TITLE
After updating to version 0.156.0 I got exception if querying empty string

### ISSUE THREAD

**Issue by wenwenchenbosch** [danielaparker/jsoncons#270]
I used before jsoncons 1.105.0 and the following code works:
`jsoncons::json pathRes = jsonpath::json_query(json_tree, "", jsonpath::result_type::path);`
as result, `pathRes.size()` is equal to 0.

After I update the version to 0.156.0. I got the following assert by running the code:
```
ERROR: Fatal runtime error: assertion 'state_stack_.size() == 2' failed at  <> :0
```

Is it a bug or feature?

**Enumerate the steps to reproduce the bug**
- compile the code above
- run it

**Include a small, self-contained example if possible**

**What compiler, architecture, and operating system?**

- Compiler:  gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
- Architecture (e.g. x86, x64) x64
- Operating system: Ubuntu

**What jsoncons library version?**

 - [x] Latest release v0.156.0
 - [ ] Other release ______
 - [ ] master

**Comment by danielaparker** [comment]
Thanks for reporting this, `pathRes` should return an empty array for this case.

**Comment by danielaparker** [comment]
Fixed in patch release 0.156.1, and on master.
